src/Compilers/VisualBasic/Portable/Parser/BlockContexts/BlockContext.vb

```diff
@@ -203,7 +203,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
         End Property
 
         Friend Sub FreeStatements()
-            _parser._pool.Free(_statements)
+            If Not _statements.IsNull Then
+                _parser._pool.Free(_statements)
+                _statements = Nothing
```

Comment:
>_statements = Nothing [](http://example.com/codeflow?start=16&length=21)

Added this so the code doesn't fall apart if FreeStatements is called multiple times